### PR TITLE
syscontainers: set SELinux label for files copied to the host

### PR DESCRIPTION
**Pull request by giuseppe**
Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

**Comment by giuseppe** [comment]
@jeremyeder this should also fix the issue we were discussing last week with files copied to the host not getting the right label

File: Atomic/rpm_host_install.py [hunk 1]
```diff
@@ -25,6 +26,10 @@ def copyfile(src, dest):
             return True
         else:
             shutil.copy2(src, dest)
```
Comment (cgwalters) [inline Atomic/rpm_host_install.py]: It's more robust to use `setfscreatecon()` to atomically create the file with the right context; see what `ostree-sepolicy.c` does.

Comment (giuseppe) [reply Atomic/rpm_host_install.py]: thanks for the hint, I'll give it a try and update the patch.  I'll need to check the "copy2" doesn't override it.

File: Atomic/rpm_host_install.py [hunk 2]
```diff
@@ -52,7 +57,7 @@ def file_checksum(path, blocksize=(1<<20)):
         return h.hexdigest()
 
     @staticmethod
-    def rm_add_files_to_host(old_installed_files_checksum, exports, prefix="/", files_template=None, values=None, rename_files=None):
+    def rm_add_files_to_host(old_installed_files_checksum, exports, prefix="/", files_template=None, values=None, rename_files=None, files_content=None):
```
[the remark this answers is not in the document]
Comment (giuseppe) [reply Atomic/rpm_host_install.py]: thanks to have noticed it.  It is a piece from another patch I was working on, I dropped it in a fixup patch

**Comment by giuseppe** [comment]
I've changed the implementation to use `setfscreatecon_raw`.  I've marked the PR as WIP as I have not added a new test yet.

@peterbaouoft could you please verify if the firewalld system container still work with the new implementation and that the new files get a correct label?

File: Atomic/rpm_host_install.py [hunk 3]
```diff
@@ -74,7 +79,14 @@ def rm_add_files_to_host(old_installed_files_checksum, exports, prefix="/", file
         # if there is a directory hostfs/ under exports, copy these files to the host file system.
         hostfs = os.path.join(exports, "hostfs")
         new_installed_files_checksum = {}
-        if os.path.exists(hostfs):
+        if not os.path.exists(hostfs):
+            return new_isntalled_files_checksum
```
[the remark this answers is not in the document]
Comment (giuseppe) [reply Atomic/rpm_host_install.py]: :/ yes thanks.  Fixed in a newer version

**Comment by giuseppe** [comment]
@peterbaouoft from my tests here, it solves the issue.  Have you had a chance to verify this on your end?

**Comment by peterbaouoft** [comment]
@giuseppe , currently checking. The last time steve and I verified the first stage `systemctl start firewalld` will be able to start without(selinux) problem after applying the patch fix.

However, recently I checked on the host I had, I am unable to run `atomic run firewalld firewall-cmd --state` saying that I get `org.fedoraproject.FirewallD1.info is not registered`. (even if systemctl start firewalld succeeds )

I am going to double check that again by provisioning a new host, so might take a while, and will update the result and log once the test is complete =).

**Comment by giuseppe** [comment]
@rhatdan are you fine with this PR?

**Comment by rhatdan** [comment]
Well this will cause the target to be labeled correctly and all of the content to have that label.  Any content created under the label may or may not be labeled correctly.  

So if you are copying a single file this will be fine, if you are copying a directory it could be a problem.

**Comment by giuseppe** [comment]
we copy each file separately so that selabel_lookup_raw+setfscreatecon_raw is done for each file.  Doesn't that solve the problem with the directories labelling?

**Comment by rhatdan** [comment]
Yes that will solve the problem.

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**